cloudinit/config/schema.py

```diff
@@ -390,48 +451,108 @@ def _get_schema_examples(schema, prefix=''):
     return rst_content
 
 
-def get_schema_doc(schema):
-    """Return reStructured text rendering the provided jsonschema.
+def get_meta_doc(meta: MetaSchema, schema: dict) -> str:
+    """Return reStructured text rendering the provided metadata.
 
-    @param schema: Dict of jsonschema to render.
-    @raise KeyError: If schema lacks an expected key.
+    @param meta: Dict of metadata to render.
+    @raise KeyError: If metadata lacks an expected key.
     """
-    schema_copy = deepcopy(schema)
-    schema_copy['property_doc'] = _get_property_doc(schema)
-    schema_copy['examples'] = _get_schema_examples(schema)
-    schema_copy['distros'] = ', '.join(schema['distros'])
+
+    # Some modules set this to falsy value, exit gracefully
+    if not meta or not schema:
+        return ""
+    keys = set(meta.keys())
+    expected = set(
+        {
+            "id",
+            "title",
+            "examples",
+            "frequency",
+            "distros",
+            "description",
+            "name",
+        }
+    )
+    if keys != expected:
+        raise KeyError(
+            "Missing module metadata key(s) {}".format(expected - keys)
+        )
```

Comment:
This will also error if additional keys in meta exist beyond what is expected with a message like "Missing module metadata key(s) set()". Is it worth differentiating the errors here a bit?
Note formatting/whitespace might be off in the "suggestion" here.
```suggestion
    error_msg = ""
    if keys.difference(expected):
        error_msg = "Additional unexpected keys found in module meta: {}".format(keys.difference(expected) 
    elif expected.difference(keys):
        error_msg = "Missing expected keys in module meta: {}".format(expected.difference(keys))
    if error_msg:
       raise KeyError(error_msg)
```